modules/creating_a_blueprint.md

```diff
@@ -0,0 +1,170 @@
+
+```javascript
+    get initializing() {
+        return {
+            myCustomInitPhaseStep() {
+                // Do all your stuff here
+            },
+            myAnotherCustomInitPhaseStep(){
+                // Do all your stuff here
+            }
+        };
+    }
+```
+
+3) Partially override a phase, this is when the blueprint gets the phase from JHipster and customizes it.
+
+```javascript
+    get initializing() {
+        const phaseFromJHipster = super._initializing();
+        const myCustomPhaseSteps = {
+            displayLogo() {
+                // override the displayLogo method from the _initializing phase of JHipster
+            },
+            myCustomInitPhaseStep() {
+                // Do all your stuff here
+            },
+        }
+        return Object.assign(phaseFromJHipster, myCustomPhaseSteps);
+    }
+```
+
+You can also access to JHipster's variables and functions directly from a Blueprint.
+
+## Available variables and functions
+
+### Variables from configuration:
+
+You can access to configuration in `.yo-rc.json` which will consist of both the JHipster config and your blueprint config.
+
+### Global variables:
+
+You can use constants in [generator-constants](https://github.com/jhipster/generator-jhipster/blob/master/generators/generator-constants.js):
+
+```
+    const javaDir = `${jhipsterConstants.SERVER_MAIN_SRC_DIR + this.packageFolder}/`;
+    const resourceDir = jhipsterConstants.SERVER_MAIN_RES_DIR;
+    const webappDir = jhipsterConstants.CLIENT_MAIN_SRC_DIR;
+```
+
+### Functions:
+
+You can use all functions in [generator-base](https://github.com/jhipster/generator-jhipster/blob/master/generators/generator-base.js):
+
+```
+    this.angularAppName = this.getAngularAppName(); // get the Angular application name.
+    this.printJHipsterLogo(); // to print the JHipster logo
+```
+
+**Note**: The functions in `generator-base.js` and variables in `generator-constants.js` are part of public API and hence will follow semver versioning. But other files like `generator-base-private.js`, `utils.js` etc will not follow semver versioning and might break method signature across minor versions.
+
+## Registering a blueprint to the JHipster marketplace
+
+To have your blueprint available in [the JHipster marketplace]({{ site.url }}/modules/marketplace/), you need to make sure you have the 2 keyword `yeoman-generator` and `jhipster-blueprint` in your published npm `package.json`.
+If you find any entry in the marketplace which is not a JHipster module or blueprint, you can help to blacklist it by adding it to the `blacklistedModules` section of the [modules-config.json file](https://github.com/jhipster/jhipster.github.io/blob/master/modules/marketplace/data/modules-config.json) by doing a Pull Request to the [jhipster/jhipster.github.io project](https://github.com/jhipster/jhipster.github.io).
+
+Your blueprint will become "verified" if the JHipster team verifies it.
```

Comment:
Would “official” be better here?

Reply:
if its third party we cant make t official, we thought of having a verification system for modules but we never did that for any module. Maybe its better to just remove this line